File: application/api/forms/value-change-event.js
/* global Java */

define(['boxing'], function(B) {
    var className = "com.eas.client.forms.events.ValueChangeEvent";
    var javaClass = Java.type(className);
    /**
     * Generated constructor.
     * @constructor ValueChangeEvent ValueChangeEvent
     */
    function ValueChangeEvent() {
        var maxArgs = 0;
        var delegate = arguments.length > maxArgs ?
              arguments[maxArgs] 
            : new javaClass();

        Object.defineProperty(this, "unwrap", {
            configurable: true,
            value: function() {
                return delegate;
            }
        });
        if(ValueChangeEvent.superclass)
            ValueChangeEvent.superclass.constructor.apply(this, arguments);
        delegate.setPublished(this);
        Object.defineProperty(this, "newValue", {
            get: function() {
                var value = delegate.newValue;
                return B.boxAsJs(value);
            }
        });

        Object.defineProperty(this, "oldValue", {
            get: function() {
                var value = delegate.oldValue;
                return B.boxAsJs(value);
            }
        });

        Object.defineProperty(this, "source", {
            get: function() {
                var value = delegate.source;
                return B.boxAsJs(value);
            }
        });

    };

    var ScriptsClass = Java.type("com.eas.script.Scripts");
    var space = ScriptsClass.getSpace();
    space.putPublisher(className, function(aDelegate) {
        return new ValueChangeEvent(aDelegate);
    });
    return ValueChangeEvent;
});

File: platypus-js-web-client/src/main/java/com/eas/ui/events/ValueChangeEvent.java
package com.eas.ui.events;

import com.eas.ui.Widget;

/**
 *
 * @author mgainullin
 */
public class ValueChangeEvent extends Event {

    protected Object oldValue;
    protected Object newValue;

    public ValueChangeEvent(Widget source, Object oldValue, Object newValue) {
        super(source, source);
        this.oldValue = oldValue;
        this.newValue = newValue;
    }

    public Object getOldValue() {
        return oldValue;
    }

    public Object getNewValue() {
        return newValue;
    }

}
